test/libraries/PoolStateLibrary.t.sol

```diff
@@ -0,0 +1,490 @@
+        uint128 liquidity = PoolStateLibrary.getLiquidity(manager, poolId);
+
+        // out of range liquidity is not added to Pool.State.liquidity
+        if (tick < _params.tickLower || tick >= _params.tickUpper) {
+            assertEq(liquidity, 0);
+        } else {
+            assertEq(liquidity, uint128(uint256(_params.liquidityDelta)));
+        }
+    }
+
+    function test_getTickBitmap() public {
+        int24 tickLower = -300;
+        int24 tickUpper = 300;
+        // create liquidity
+        modifyLiquidityRouter.modifyLiquidity(
+            key, IPoolManager.ModifyLiquidityParams(tickLower, tickUpper, 10_000 ether), ZERO_BYTES
+        );
+
+        (int16 wordPos, uint8 bitPos) = TickBitmap.position(tickLower / key.tickSpacing);
+        uint256 tickBitmap = PoolStateLibrary.getTickBitmap(manager, poolId, wordPos);
+        assertNotEq(tickBitmap, 0);
+        assertEq(tickBitmap, 1 << bitPos);
+
+        (wordPos, bitPos) = TickBitmap.position(tickUpper / key.tickSpacing);
+        tickBitmap = PoolStateLibrary.getTickBitmap(manager, poolId, wordPos);
+        assertNotEq(tickBitmap, 0);
+        assertEq(tickBitmap, 1 << bitPos);
+    }
+
+    function test_fuzz_getTickBitmap(IPoolManager.ModifyLiquidityParams memory params) public {
+        (IPoolManager.ModifyLiquidityParams memory _params,) =
+            Fuzzers.createFuzzyLiquidity(modifyLiquidityRouter, key, params, ZERO_BYTES);
+
+        (int16 wordPos, uint8 bitPos) = TickBitmap.position(_params.tickLower / key.tickSpacing);
+        (int16 wordPosUpper, uint8 bitPosUpper) = TickBitmap.position(_params.tickUpper / key.tickSpacing);
+
+        uint256 tickBitmap = PoolStateLibrary.getTickBitmap(manager, poolId, wordPos);
+        assertNotEq(tickBitmap, 0);
+
+        // in fuzz tests, the tickLower and tickUpper might exist on the same word
+        if (wordPos == wordPosUpper) {
+            assertEq(tickBitmap, (1 << bitPos) | (1 << bitPosUpper));
+        } else {
+            assertEq(tickBitmap, 1 << bitPos);
+        }
+    }
+
+    function test_getPositionInfo() public {
+        // create liquidity
+        modifyLiquidityRouter.modifyLiquidity(
+            key, IPoolManager.ModifyLiquidityParams(-60, 60, 10_000 ether), ZERO_BYTES
+        );
+
+        // swap to create fees, crossing a tick
+        uint256 swapAmount = 10 ether;
+        swap(key, true, -int256(swapAmount), ZERO_BYTES);
+        (, int24 currentTick,,) = PoolStateLibrary.getSlot0(manager, poolId);
+        assertNotEq(currentTick, -139);
+
+        // poke the LP so that fees are updated
+        modifyLiquidityRouter.modifyLiquidity(key, IPoolManager.ModifyLiquidityParams(-60, 60, 0), ZERO_BYTES);
+
+        bytes32 positionId = keccak256(abi.encodePacked(address(modifyLiquidityRouter), int24(-60), int24(60)));
+
+        (uint128 liquidity, uint256 feeGrowthInside0X128, uint256 feeGrowthInside1X128) =
+            PoolStateLibrary.getPositionInfo(manager, poolId, positionId);
```

Comment:
we should make a PositionId type (similar to PoolId) so that users of the extsload library dont need to manually hash the position Id